common/io.cpp

```diff
@@ -234,6 +234,17 @@ ExpectedSize StreamReader::Read(vector<uint8_t>::iterator start, vector<uint8_t>
 	return is_->gcount();
 }
 
+expected::ExpectedSize FileSize(const string &path) {
+	// Probably not as efficient as stat(), but portable.
+	ifstream is(path, ifstream::ate | ifstream::binary);
+	if (is.bad()) {
```

Comment:
Q: Is this the same as `(!is)`?

Reply:
No. But maybe it's better to use it, since it catches more errors. It's annoying that the bool operator does not have a named equivalent.

Reply:
Ok, fixed this, and the one above, which is the one I took it from. I made this into a new commit, since I'm still not 100% sure about it and I'd like @vpodzime to comment.

Reply:
Yeah, the bool operator covers more cases, AFAICT. I keep forgetting the gore details, though.